File: password-generator.js
#!/usr/bin/env node

function generatePassword(length = 12, useDigits = true, useSpecial = true) {
    let chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
    if (useDigits) chars += '0123456789';
    if (useSpecial) chars += '!@#$%^&*()_+~`|}{[]:;?><,./-=';
    let password = '';
    for (let i = 0; i < length; i++) {
        const randomIndex = Math.floor(Math.random() * chars.length);
        password += chars[randomIndex];
    }
    return password;
}

// Разбор аргументов командной строки
const args = process.argv.slice(2);
let length = 12;
let noDigits = false;
let noSpecial = false;

for (let i = 0; i < args.length; i++) {
    if (args[i] === '-l' && args[i+1]) {
        length = parseInt(args[i+1]);
        i++;
    } else if (args[i] === '--no-digits') {
        noDigits = true;
    } else if (args[i] === '--no-special') {
        noSpecial = true;
    }
}

const password = generatePassword(length, !noDigits, !noSpecial);
console.log(password);

File: password-generator.go
package main

import (
    "flag"
    "fmt"
    "math/rand"
    "time"
)

func generatePassword(length int, useDigits, useSpecial bool) string {
    letters := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
    digits := "0123456789"
    special := "!@#$%^&*()_+~`|}{[]:;?><,./-="

    charset := letters
    if useDigits {
        charset += digits
    }
    if useSpecial {
        charset += special
    }

    rand.Seed(time.Now().UnixNano())
    password := make([]byte, length)
    for i := range password {
        password[i] = charset[rand.Intn(len(charset))]
    }
    return string(password)
}

func main() {
    length := flag.Int("l", 12, "длина пароля")
    noDigits := flag.Bool("no-digits", false, "исключить цифры")
    noSpecial := flag.Bool("no-special", false, "исключить спецсимволы")
    flag.Parse()

    password := generatePassword(*length, !*noDigits, !*noSpecial)
    fmt.Println(password)
}
